src/__tests__/third-party/baseweb/pages/_document.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import Document, {
+  DocumentContext,
+  Head,
+  Html,
+  Main,
+  NextScript,
+} from 'next/document';
+import { Provider as StyletronProvider } from 'styletron-react';
+import { Server, Sheet } from 'styletron-engine-atomic';
+import type { RenderPageResult } from 'next/dist/next-server/lib/utils';
+import { styletron } from '../styletron';
+
+type Props = RenderPageResult & {
+  stylesheets: Sheet[];
+};
+
+export default class CustomDocument extends Document<Props> {
+  static async getInitialProps(ctx: DocumentContext) {
+    const page = ctx.renderPage((App) => (props) => (
+      <StyletronProvider value={styletron}>
+        <App {...props} />
+      </StyletronProvider>
+    ));
+
+    const stylesheets = (styletron as Server).getStylesheets() || [];
```

Comment:
For some reason `styletron` is undefined here and this crashes 💥

Reply:
One potential issue might come from the fact that we render the [custom document in `client` mode](https://github.com/toomuchdesign/next-page-tester/blob/3512c8fd2f97cce413181fbdb4d82bba5d856c7f/src/_document/render.tsx#L26) since the server version doesn't currently work. It's the opposite way Next.js deals with document component: rendered once on the server as string and never rehydrated.

I see that `styletron` file makes some environment detection, but only the client version is currently used, since `next-page-tester` renders custom document in client environment (while Next.js does the opposite):
https://github.com/toomuchdesign/next-page-tester/blob/b1e4df6ef251d7a8d15cc45bb145ef40cce2f477/src/__tests__/third-party/baseweb/styletron.ts#L9

The best solution would consist of rendering custom document as Next.js does. More or less like:
- Render document element in server environement
- Output it as string
- Remove the contents of root div `#__next` element
- Mount app component as a child of `#__next` element

This would also maybe fix the issue described here, but I haven't been able to get there at the moment.

If this is the problem, could it be temporarily addressed in `baseweb/styletron.ts`?

Reply:
I kept that file as it is intentionally. Almost everyone using that library will use it like this. It's from a boilerplate. Point being, users should not be forced to change their code to use this library.